test/llvm-intrinsics/bitreverse_small_type.ll

```diff
@@ -0,0 +1,143 @@
+;; Check that llvm.bitreverse.* intrinsics are lowered to emulation functions for
+;; 2/4-bit types.
+
+; The scalar calls can be generated from:
+;
+;   using uint2_t = unsigned _BitInt(2);
+;   using uint4_t = unsigned _BitInt(4);
+;   uint4_t reverse4(uint4_t a) {
+;   a = ((0x5 & a) << 1) | (0x5 & (a >> 1));
+;   return (a << 2) | (a >> 2);
+;   }
+;   uint2_t reverse2(uint2_t a) {
+;   return (a << 1) | (a >> 1);
+;   }
+;
+; with the command clang++ -O2 -emit-llvm -S bitreverse.c.
+; The vector cases cannot be generated by clang++ since elements smaller than a char
+; are not supported in vectors.  The vector cases are lowered for completeness.
```

Comment:
```suggestion
; The vector cases (v4i2, v4i4 etc) cannot be generated by clang++ since vectors containing BitInt elements smaller than a char
; are not supported in clang.  In this test, such vector cases have been added manually tfor completeness.
```

Reply:
Updated comment